File: demos/api/open.py
#!/usr/bin/env python

import gmsh
import sys

if len(sys.argv) < 2:
    print "Usage: " + sys.argv[0] + " file.geo [options]"
    exit(0)

gmsh.initialize()
gmsh.optionSetNumber("General.Terminal", 1)
gmsh.open(sys.argv[1])
gmsh.modelMeshGenerate(3)
gmsh.write("test.msh")
gmsh.finalize()

File: demos/api/open.cpp
#include <iostream>
#include <gmsh.h>

int main(int argc, char **argv)
{
  if(argc < 2){
    std::cout << "Usage: " << argv[0] << " file.geo [options]" << std::endl;
    return 1;
  }

  gmshInitialize();
  gmshOptionSetNumber("General.Terminal", 1);
  gmshOpen(argv[1]);
  gmshModelMesh(3);
  gmshExport("test.msh");
  gmshFinalize();

  return 0;
}
